File: src/main/kotlin/kr/co/programmers/P12939.kt
package kr.co.programmers

// https://github.com/antop-dev/algorithm/issues/390
class P12939 {
    fun solution(s: String): String {
        val sb = StringBuilder().apply {
            append(" ")
            append(s)
            reverse()
        }

        var min = Int.MAX_VALUE
        var max = Int.MIN_VALUE
        var num = 0 // 만들어지는 수
        var digit = 1 // 자리수 곱샘
        for (ch in sb) {
            when (ch) {
                in '0'..'9' -> { // 숫자
                    num += (ch - '0') * digit
                    digit *= 10
                }
                '-' -> { // 부호
                    num *= -1
                }
                else -> { // 공백
                    min = minOf(min, num)
                    max = maxOf(max, num)
                    num = 0
                    digit = 1
                }
            }
        }
        return "$min $max"
    }
}

File: src/python3/kr/co/programmers/p12939.py
import sys


def solution(s):
    _min = sys.maxsize
    _max = -sys.maxsize - 1

    v = 0
    digit = 1
    for i in reversed(range(len(s))):
        ch = s[i]

        if ch != ' ':
            if ch == '-':
                v = v * -1
            else:
                v = v + (ord(ch) - ord('0')) * digit
                digit = digit * 10

        if ch == ' ' or i == 0:
            _min = min(_min, v)
            _max = max(_max, v)
            v = 0
            digit = 1

    return f"{_min} {_max}"
